proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,532 @@
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+
+import pytest
+
+from pyomo.environ import (ConcreteModel,
+                           Set,
+                           Param,
+                           Var,
+                           Expression,
+                           Constraint,
+                           value,
+                           units as pyunits,
+                           SolverStatus)
+from pyomo.util.check_units import assert_units_consistent, check_units_equivalent
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import (degrees_of_freedom,
+                                              number_variables,
+                                              number_total_constraints,
+                                              number_unused_variables)
+from idaes.core.util.scaling import (calculate_scaling_factors,
+                                     unscaled_variables_generator,
+                                     unscaled_constraints_generator,
+                                     badly_scaled_var_generator)
+from idaes.core.util.exceptions import (PropertyPackageError,
+                                        PropertyNotSupportedError,
+                                        ConfigurationError)
+from idaes.core.util import get_solver
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+solver.options["nlp_scaling_method"] = "user-scaling"
+
+
+# -----------------------------------------------------------------------------
+class PropertyUnitTestHarness(object):
+    def configure_class(self, m):
+        self.configure()
+
+        # attaching objects to model to use in pytest frame
+        obj_list = ['component_dict', 'phase_dict', 'param_obj_dict',
+                    'default_scaling_factor_dict', 'stateblock_obj_dict', 'state_vars_list',
+                    'statistics_dict', 'general_methods_dict', 'default_methods_dict']
+        for obj_str in obj_list:
+            if not hasattr(self, obj_str):
+                raise ConfigurationError(
+                    "{obj_str} is a required attribute for the PropertyUnitTestHarness. "
+                    "Add {obj_str} to the configure function that sets up the test harness.".
+                        format(obj_str=obj_str))
+            obj = getattr(self, obj_str)
+            setattr(m, obj_str, obj)
```

Comment:
I agree with @andrewlee94, and it doesn't seem we need to do this anymore. Or maybe we should figure out how to re-write this harness so it isn't necessary.

Failing that, we could put these attributes on a single Block (named `_proteuslib_prop_test`?) which we could check to ensure doesn't clash with an existing name on the model.